kedro/framework/hooks/specs.py

```diff
@@ -240,41 +240,45 @@ class DatasetSpecs:
     """Namespace that defines all specifications for a dataset's lifecycle hooks."""
 
     @hook_spec
-    def before_dataset_loaded(self, dataset_name: str) -> None:
+    def before_dataset_loaded(self, dataset_name: str, node: Node) -> None:
         """Hook to be invoked before a dataset is loaded from the catalog.
 
         Args:
             dataset_name: name of the dataset to be loaded from the catalog.
+            node: node: The ``Node`` to ran.
```

Comment:
I made some more changes to the documentation, the previous one is buggy and won't even print the log. I also considered updating the example for `MemoryProfilingHooks`, but decided not as that example is purely about dataset profiling thus the node argument isn't useful.

Couple changes
* `logging.getLogger(self._class.__name__)` -> `logging.getLogger(__name__)` - this makes sure the log will be printed properly with the default `logging.yml`
* Instead of printing the absolute timestamp, the hook print how many time it takes to load a dataset instead.